src/actions/delete-environment/index.ts

```diff
@@ -1,41 +1,32 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+import { deleteEnvironment } from "@microsoft/powerplatform-cli-wrapper/dist/actions";
 import * as core from '@actions/core';
-import { AuthKind, AuthHandler, DefaultRunnerFactory, RunnerFactory} from '../../lib';
+import { YamlParser } from '../../lib/parser/YamlParser';
+import { ActionsHost } from '../../lib/host/ActionsHost';
+import getCredentials from "../../lib/auth/getCredentials";
+import { runnerParameters } from '../../lib/runnerParameters';
 
 (async () => {
     if (process.env.GITHUB_ACTIONS) {
-        await main(DefaultRunnerFactory);
+        await main();
     }
-})();
+})().catch(error => {
+    const logger = runnerParameters.logger;
+    logger.error(`failed: ${error}`);
+    core.endGroup();
+});
 
-export async function main(factory: RunnerFactory): Promise<void> {
-    let pac;
-    try {
-        core.startGroup('delete-environment:');
-        const envUrl = core.getInput('environment-url', { required: false });
-        const envId = core.getInput('environment-id', { required: false });
+export async function main(): Promise<void> {
+    const taskParser = new YamlParser();
+    const parameterMap = taskParser.getHostParameterEntries('delete-environment');
 
-        let deleteEnvArgs;
-        if (envUrl) {
-            deleteEnvArgs = ['admin', 'delete', '--url', envUrl];
-        } else if (envId) {
-            deleteEnvArgs = ['admin', 'delete', '-id', envId];
-        } else {
-            throw new Error(
-                "Must provide either environment-id or environment-url to delete an environment!"
-            );
-        }
-
-        pac = factory.getRunner('pac', process.cwd());
-        await new AuthHandler(pac).authenticate(AuthKind.ADMIN);
-        await pac.run(deleteEnvArgs);
-        core.info('environment deleted');
-        core.endGroup();
-    } catch (error) {
-        core.setFailed(`failed: ${error.message}`);
-        throw error;
-    } finally {
-        await pac?.run(["auth", "clear"]);
-    }
+    core.startGroup('delete-environment:');
+    await deleteEnvironment({
+        credentials: getCredentials(),
+        environmentUrl: parameterMap['environment-url'],
```

Comment:
your code is still breaking existing users! Your PR only passes since you also changed the PR workflow
-> you must add the parameter for env-id; removing it now would be a breaking change -> BAD